challenges-services/src/main/java/org/exoplatform/challenges/storage/ChallengeStorage.java

```diff
@@ -0,0 +1,41 @@
+package org.exoplatform.challenges.storage;
+
+import org.exoplatform.challenges.dao.ChallengeDAO;
+import org.exoplatform.challenges.entity.ChallengeEntity;
+import org.exoplatform.challenges.model.Challenge;
+import org.exoplatform.challenges.utils.EntityMapper;
+import org.exoplatform.challenges.utils.Utils;
+import org.exoplatform.social.core.identity.model.Identity;
+
+public class ChallengeStorage {
+
+  private ChallengeDAO challengeDAO;
+
+  public ChallengeStorage(ChallengeDAO challengeDAO) {
+    this.challengeDAO = challengeDAO;
+  }
+
+  public Challenge saveChallenge(Challenge entity, String username) throws IllegalArgumentException {
```

Comment:
Could you change the DTO name to challenge instead of entity for better readability

Reply:
ok